src/main/kotlin/org/cascadebot/cascadebot/data/objects/donation/Tier.kt

```diff
@@ -0,0 +1,183 @@
+
+    private constructor()
+
+    constructor(flags: MutableSet<Flag>) :
+            this(
+                    null,
+                    flags,
+                    ArrayList()
+            )
+
+    constructor(parent: String?, flags: MutableSet<Flag>, extras: MutableList<TierExtra>) {
+        this.parent = parent
+        this.flags = flags
+        this.extras = extras
+    }
+
+    fun addFlag(flag: Flag) {
+        flags.add(flag)
+    }
+
+    fun getParent(): Tier? {
+        return if (parent == null) null else tiers[parent]
+    }
+
+    fun getAllFlags(): Set<Flag> {
+        val flags = flags.toMutableSet()
+        if (getParent() != null) {
+            flags.addAll(getParent()!!.getAllFlags())
+        }
+        return flags
+    }
+
+    override fun getFlag(id: String): Flag? {
+        var returnFlag = flags.stream().filter { flag: Flag -> flag.id == id }.findFirst().orElse(null)
+        if (parent != null && returnFlag == null) {
+            returnFlag = tiers[parent]!!.getFlag(id)
+        }
+        return returnFlag
+    }
+
+    fun isTierParent(tier: String): Boolean {
+        if (StringUtils.isBlank(parent)) {
+            return false
+        }
+        return if (parent == tier) {
+            true
+        } else tiers[parent]!!.isTierParent(tier)
+    }
+
+    /**
+     * Returns the benefits gave in this tier to a guild.
+     *
+     * @param locale The locale to use for translations
+     * @param idsUsed This parameter should be null. It's used for when calling parent method. It stores flags that have been used already.
+     * @return The guild benefits gave in this tier
+     */
+    fun getGuildTierString(locale: Locale, idsUsed: MutableList<String>): String {
+        val tierStringBuilder = StringBuilder()
+        for (flag in flags) {
+            if (!idsUsed.contains(flag.id)) {
+                if (flag.scope == FlagScope.GUILD) {
+                    idsUsed.add(flag.id)
+                    tierStringBuilder.append(" - **").append(flag.getName(locale)).append(":** ").append(flag.getDescription(locale)).append('\n')
+                }
+            }
+        }
+        for (extra in extras) {
+            if (extra.scope == FlagScope.GUILD) { //Ignore t-shirt related extra stuff as that isn't a guild thing.
+                val extraStr = getLanguage(locale)!!.getString(extra.path).orElse("No language string defined")
+                tierStringBuilder.append(" - ").append(extraStr).append('\n')
+            }
+        }
+        if (parent != null && parent != "default") {
+            tierStringBuilder.apply {
+                append('\n')
+                append("**__Inherited from ").append(parent).append(":__**\n")
+                append(tiers[parent]!!.getGuildTierString(locale, idsUsed))
+            }
+        }
+        return tierStringBuilder.toString()
+    }
+
+    class TierExtra {
+        @Getter
+        val path: String?
+
+        @Getter
+        val scope: FlagScope?
```

Comment:
Remove getters please (Whoops I missed this). Make sure to remove the import as well